tests/ssl-opt.sh

```diff
@@ -1008,7 +1008,21 @@ run_test    "Unique IV in GCM" \
             -u "IV used" \
             -U "IV used"
 
-# Test for context-specific CRT verification callback
+# Tests for certificate verification callback
+run_test    "Configuration-specific CRT verification callback" \
+            "$P_SRV debug_level=3" \
+            "$P_CLI context_crt_cb=0 debug_level=3" \
+            0 \
+            -s "Protocol is TLSv1.2" \
+            -s "Ciphersuite is TLS-ECDHE-RSA-WITH-CHACHA20-POLY1305-SHA256" \
```

Comment:
Is this a bit fragile? If the list of default ciphersuites change, won't this also need changing?

Reply:
Yes, that's a general problem we have with `ssl-opt.sh` - the test template is taken from the `Default` test, which has the same check. However, there's no reason not to remove this line in this test, to not introduce yet another test which isn't compatible with reduced configs.

Reply:
There are many problems with `ssl-opt.sh` but I don't think this is one of them. IMO it's rather a problem of insufficient editing after copy-paste, which can happen with any script and to anyone :)

Reply:
@mpg I didn't mean to defend the copy-paste - it should be changed - but the `Default` test _does_ have the same problem, doesn't it? It hardcodes the expectation of ChaChaPoly as the chosen ciphersuite.

Reply:
Yes, the default test clearly requires a bunch more `requires_config`, as do most tests here, which is a big problem with `ssl-opt.sh` indeed.

However the way I see it, the hardcoding it intentional here: we want to test what our default behaviour is (hence the "Default" in the test name), and the alternative to hardcoding is to have the test system also compute what the selected algorithms should be based on `config.h`, but IMO that's fragile in another way: what if we make the same mistake in computing those in the test script and in the library? Hardcoding, with all the issues it has, is more robust in that respect.

But that's largely out of scope here, and probably a discussion we'll need to have some other time.